### ISSUE TITLE
Clarify the status of an uploaded document with errors and see about adding transactions

### ISSUE THREAD

**Issue by tmtsoftwareadmin** [tmtsoftware/icd#17]
Database updates should probably be done with transactions.  Not sure what is being done currently. MongoDB seems to support transactions and 2 phase commit.

We also need to clarify the use/status of the uploaded set of documents that have not been published.

**Comment by abrighton** [comment]
Transactions are difficult (or impossible) to implement in MongoDB, especially since we are using multiple collections. The current icd version now validates all ICD files before ingesting any into the database. If the files are all valid, there should not normally be any database errors.

**Comment by tmtsoftwareadmin** [comment]
I don't think we should close this.  It's a good idea to support transactions, maybe we need a different solution.  I prefer that we leave this open for now.
